core/conn/unixodbc/odbc/odbcclient/unixcli/cli/cdesc.cpp

```diff
@@ -2381,7 +2381,8 @@ SQLRETURN CDesc::BuildValueList(CStmt *pHandle,  //AMR - changed this pointer ty
 
 				if(IPDDescRecPtr->m_ODBCDataType == SQL_VARCHAR ||
 				   IPDDescRecPtr->m_ODBCDataType == SQL_LONGVARCHAR ||
-				   IPDDescRecPtr->m_ODBCDataType == SQL_CHAR)
+				   IPDDescRecPtr->m_ODBCDataType == SQL_CHAR &&
+                   IPDDescRecPtr->m_SQLDataType != SQLTYPECODE_BOOLEAN)
```

Comment:
Is this the operator precedence you want? As written, this code means:
if ( VARCHAR or LONG VARCHAR or ( CHAR and not BOOLEAN) )
Perhaps you meant,
if ( ( VARCHAR or LONG VARCHAR or CHAR) and BOOLEAN) ) ?